complete/src/main/java/hello/Application.java

```diff
@@ -13,7 +13,10 @@
 public class Application extends AsyncConfigurerSupport {
 
     public static void main(String[] args) {
-        SpringApplication.run(Application.class, args);
+        SpringApplication.run(Application.class, args).close();
+
+        // Close application
+        System.exit(0);
     }
```

Comment:
It would be better to close the application context. In either case it seems like you need to wait for something to happen before you do that? So we might need a latch or something that can block the main thread until it happens. Or maybe we could pose the question another way.